data_collection/gazette/spiders/base.py

```diff
@@ -188,3 +188,85 @@ def get_gazette(self, document):
             file_urls=(document[0],),
             power="executive",
         )
+
+
+class DoemGazetteSpider(BaseGazetteSpider):
+    """
+    Base spider for all cities listed on https://doem.org.br
+    """
+
+    allowed_domains = ["doem.org.br"]
+    start_date = date(2009, 1, 1)
+
+    def start_requests(self):
+        yield scrapy.Request(self.get_url(), callback=self.parse_pagination)
+
+    def parse_pagination(self, response):
+        """
+        This parse function is used to get all the pages available and
+        return request object for each one
+        """
+        return [
+            scrapy.Request(self.get_url(page), callback=self.parse)
+            for page in range(1, 1 + self.get_last_page(response))
+        ]
+
+    def parse(self, response):
+        """
+        Parse each page from the results page and yield the gazette issues available.
+        """
+        GAZETTE_CONTAINER_CSS_QUERY = "div.box-diario"
+        gazette_boxes = response.css(GAZETTE_CONTAINER_CSS_QUERY)
+
+        for gazette_box in gazette_boxes:
+            file_url = self.get_pdf_url(gazette_box)
+            date = self.get_gazette_date(gazette_box)
+            edition_number = self.get_edition_number(gazette_box)
+            yield Gazette(
+                date=date,
+                file_urls=[file_url],
+                edition_number=edition_number,
+                power="executive_legislative",
+            )
+
+    def get_url(self, page=None):
+        state, city = self.name.split("_")
```

Comment:
Just a comment about cities with multiword names, they don't use the same pattern we use here. All words are joined together without `"_"`. Consider if you want to change it now or change it later.

One thing to think about: I personally don't like to rely on the spider name to make anything in the crawl. This is mainly by one bad experience after changing naming patterns for the project and having some spiders break down unnecessarily. For me, making a minor "code repetition" is valid. What do you think?

Reply:
Excelent points. Explicit is better than implicit :) 
I'll make the city name as expected to build the URL an attribute of the spider instance, to be provided by the person building the specific child-spider.